### ISSUE TITLE
handle DOMAIN push option

### ISSUE THREAD

**Issue by fkooman** [Amebis/eduVPN#110]
```
              DOMAIN name -- Set Connection-specific DNS Suffix.
```

The VPN server can push this, multiple times, for example I setup vpn.tuxed.net to send two domains:

```
push "dhcp-option DOMAIN tuxed.net"
push "dhcp-option DOMAIN tuxed.nl"
```

It seems OpenVPN for Windows (and eduVPN/LC) only takes the last one and uses that as the "connection specific DNS suffix", the other one is ignored.

In Viscosity the first one is used as the "connection specific DNS suffix", the rest as "search domains". On Linux (NetworkManager) they are all used as search domains.

Would it be possible to get this behavior also in OpenVPN/eduVPN/LC? It seems there used to be some time ago a `DOMAIN-SEARCH` option, but seems this is no longer available...

**Comment by rozmansi** [comment]
This is OpenVPN's business actually. This needs to be addressed in the https://github.com/OpenVPN/openvpn repo. I am sure other OpenVPN deployments would benefit from this too.

Windows does allow to have multiple search domains assigned to an interface. If I recall correctly, OpenVPN is shelling out the `netsh.exe` to set various interface parameters. I am not familiar if there were any past efforts to support multiple DNS sufixes in OpenVPN or what went wrong it doesn't support it yet.

I am keeping this open, should I ever be able to find some time to address it. I'll ask the community on 2019 Hackathon.

**Comment by gvde** [comment]
> This is OpenVPN's business actually. This needs to be addressed in the https://github.com/OpenVPN/openvpn repo. I am sure other OpenVPN deployments would benefit from this too.

Handling those options for the connection is OpenVPNs. Providing these options in the configuration is eduvpn. 

> Windows does allow to have multiple search domains assigned to an interface. If I recall correctly, OpenVPN is shelling out the `netsh.exe` to set various interface parameters. I am not familiar if there were any past efforts to support multiple DNS sufixes in OpenVPN or what went wrong it doesn't support it yet.

Windows does actually allow you to configure multiple search domains in the advanced settings for the IPv4 protocol of an interface. On the same page you can also set the dns suffix for the connection.

**Comment by fkooman** [comment]
See also Jan Just's mail(s) on the openvpn-devel list 

Re: [Openvpn-devel] [Openvpn-users] Multiple DNS search suffixes on Windows
3/5/20, 1:53 PM

and follow up.

**Comment by rozmansi** [comment]
> See also Jan Just's mail(s) on the openvpn-devel list
> 
> Re: [Openvpn-devel] [Openvpn-users] Multiple DNS search suffixes on Windows
> 3/5/20, 1:53 PM
> 
> and follow up.

I looked into it. Yes, it needs polishing to make it handle the OpenVPN pushed settings the same way as other platforms do.

Unfortunately, his patch applies to Windows 10 1809 or later, and DHCP with `--dev tap` use case only. We'd still need to research other ways to set search domains programmatically: netsh, registry... Otherwise, it won't work on other Windows, with `--dev tun` (including Wintun)...

**Comment by fkooman** [comment]
This is now implemented in OpenVPN 2.5, so that should solve this issue?
